File: leet-code/numJewelsInStones.js
/**
 * @param {string} J
 * @param {string} S
 * @return {number}
 */
var numJewelsInStones = function (J, S) {
  const map = new Map()
  for (let i = 0; i < J.length; i++) {
    map.set(J[i], i)
  }
  let count = 0
  for (let i = 0; i < S.length; i++) {
    if(map.has(S[i])){
      count++
    }    
  }
  return count
}

File: leet-code/numJewelsInStones.go
package main

func numJewelsInStones(J string, S string) int {
	sMap := make(map[byte]int)
	count := 0
	for i := range J {
		sMap[J[i]] = i
	}

	for i := range S {
		if _, ok := sMap[S[i]]; ok {
			count++
		}
	}
	return count
}

func numJewelsInStones2(J string, S string) int {
	count := 0
	for i := range J {
		for j := range S {
			if J[i] == S[j] {
				count++
			}
		}
	}
	return count
}
